lib/utils.js

```diff
@@ -563,33 +563,38 @@ function isHiddenOnUnix(pathname) {
  * @public
  * @memberof Mocha.utils
  * @param {string} filepath - Base path to start searching from.
- * @param {string[]} extensions - File extensions to look for.
- * @param {boolean} recursive - Whether to recurse into subdirectories.
+ * @param {string[]} [extensions=[]] - File extensions to look for.
+ * @param {boolean} [recursive=false] - Whether to recurse into subdirectories.
  * @return {string[]} An array of paths.
  * @throws {Error} if no files match pattern.
  * @throws {TypeError} if `filepath` is directory and `extensions` not provided.
  */
 exports.lookupFiles = function lookupFiles(filepath, extensions, recursive) {
   extensions = extensions || [];
+  recursive = recursive || false;
```

Comment:
Don't change the input argument values unless they are not provided (i.e., `undefined`).

Reply:
Any falsy input (0, "", null, undefined) or no argument provided (=undefined) will be changed to its default value. What should be wrong about changing those falsy values?